### ISSUE TITLE
403 Client Error: Forbidden for url: http://hassio/snapshots

### ISSUE THREAD

**Issue by rsuplido** [danielwelch/hassio-dropbox-sync#11]
Any idea why I'm getting this error? I have keep_last set to 3.

```
[Info] Received message with command upload
[Info] Uploading all .tar files in /backup (skipping those already in Dropbox)
 > Skipping already existing file "/6dfda24c.tar"
[Info] keep_last option is set, cleaning up files...
Traceback (most recent call last):
  File "/keep_last.py", line 47, in <module>
    main(args.number)
  File "/keep_last.py", line 15, in main
    snapshot_info.raise_for_status()
  File "/usr/lib/python3.6/site-packages/requests/models.py", line 935, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 403 Client Error: Forbidden for url: http://hassio/snapshots
```

**Comment by DavidFW1960** [comment]
I am seeing this since 0.79.0 as well
```
{"command": "upload"}
[Info] Received message with command upload
[Info] Uploading all .tar files in /backup (skipping those already in Dropbox)
 > Skipping already existing file "/home-assistant-backups/26065462.tar"
 > Uploading "/backup/725e826b.tar" to "/home-assistant-backups/725e826b.tar"... DONE
 > Skipping already existing file "/home-assistant-backups/773aab24.tar"
 > Skipping already existing file "/home-assistant-backups/91c1f4a0.tar"
 > Skipping already existing file "/home-assistant-backups/a41d76e7.tar"
 > Skipping already existing file "/home-assistant-backups/c7d07f53.tar"
[Info] keep_last option is set, cleaning up files...
Traceback (most recent call last):
  File "/keep_last.py", line 47, in <module>
    main(args.number)
  File "/keep_last.py", line 15, in main
    snapshot_info.raise_for_status()
  File "/usr/lib/python3.6/site-packages/requests/models.py", line 935, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 403 Client Error: Forbidden for url: http://hassio/snapshots
```

**Comment by danielwelch** [comment]
Looks like a permissions/authentication problem, I'm guessing stemming from all the recent changes to auth in homeassistant. Taking a look at the developer docs, it looks like maybe [this is new](https://developers.home-assistant.io/docs/en/hassio_addon_communication.html#hassio-api):

> #### Hass.io API
>To enables calls to the Hass.io API, add `hassio_api: true` to config.json and read the environment variable HASSIO_TOKEN. Now you can use the API over the URL: http://hassio/. Use the HASSIO_TOKEN with header X-HASSIO-KEY. It could be that you need also change the Hass.io API role like hassio_role: default.

> Add-ons can call some API commands without need set hassio_api: true:

> /homeassistant/api
> /homeassistant/websocket
> /homeassistant/stream
> /addons/self/...
> /services*

Can you try adding `hassio_api: true` to your config.json and see if that works?

**Comment by rsuplido** [comment]
I don't think config.json is in hassio.

**Comment by DavidFW1960** [comment]
Yeah it's only since 0.79.x that I've been seeing it.

**Comment by danielwelch** [comment]
What version of hassio are you using?

**Comment by rsuplido** [comment]
I’m running 0.79.3

**Comment by DavidFW1960** [comment]
same here. 0.79.3

**Comment by danielwelch** [comment]
Not home assistant, hassio

**Comment by danielwelch** [comment]
Could this be because of the new Hassio roles security system? It looks like add-one need to have the “manager” role to call snapshot endpoints in the API. I don’t see any docs on this system so I’m just interpreting from the source code. 

https://github.com/home-assistant/hassio/blob/dev/hassio/api/security.py

**Comment by DavidFW1960** [comment]
I have no idea.. might need to take it up with awarecan or balloob on the forum/discord

**Comment by d0ugal** [comment]
I wondered if you had the same issue, and found that you did. I then found the solution, it is just a simple config.json addition. https://github.com/d0ugal/hassio-dropbox-upload/commit/763e1af11a25343ab3b05dc1dcd565738be0a116

**Comment by DavidFW1960** [comment]
I just edited the config.json (I'm running Hassio in docker on debian so I can access the files) I edited it to show:
```
{
  "name": "Dropbox Sync",
  "version": "1.2.0",
  "slug": "dropbox_sync",
  "description": "Upload your Hass.io backups to Dropbox",
  "url": "https://github.com/danielwelch/hassio-dropbox-sync",
  "startup": "before",
  "stdin": true,
  "hassio_api": true,
  "hassio_role": "manager",
  "boot": "auto",
  "map": ["backup", "share"],
  "options": {
    "oauth_access_token": "<YOUR_ACCESS_TOKEN>",
    "output": ""
  },
  "schema": {
    "oauth_access_token": "str",
    "output": "str",
    "keep_last": "int(0,)?",
    "filetypes": "str?"
  },
  "image": "dwelch2101/dropbox-sync-{arch}"
}
```
I'll report back if it works...... (I added the manager role as per @d0ugal

**Comment by danielwelch** [comment]
@DavidFW1960 I added the same line and released the 1.3.0 update two days ago. If anyone can confirm that it's working, that'd be great and I'll close this.

**Comment by rsuplido** [comment]
Can't find 1.3.0 -- still says 1.2.0 as the latest.

**Comment by danielwelch** [comment]
Sorry, forgot to update the repository version. Just pushed that update and updated the schema over there, so hopefully it should show up.

I've moved to a single-repository-per-addon method for newer add-ons so forgot about this step.

**Comment by DavidFW1960** [comment]
Thanks Daniel! Yeah my edit didn't work.... just seeing if it's fixed with 1.3 now. Thanks.
